scrabble.js

```diff
@@ -1,8 +1,213 @@
-var Scrabble = function() {};
+/* SCRABBLE */
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+var Scrabble = function() {
+
+};
+
+var LETTERS = {
+  A: 1, B: 3, C: 3,
+  D: 2, E: 1, F: 4,
+  G: 2, H: 4, I: 1,
+  J: 8, K: 5, L: 1,
+  M: 3, N: 1, O: 1,
+  P: 3, Q: 10, R: 1,
+  S: 1, T: 1, U: 1,
+  V: 4, W: 4, X: 8,
+  Y: 4, Z: 10
+};
+
+Scrabble.score = function(word) {
+  var score = 0;
+  var userWord = word.toUpperCase();
+
+  if (userWord.length == 7) {
+    score += 50;
+  }
+
+  for (var i = 0; i < userWord.length; i++) {
+    for (var letter in LETTERS) {
```

Comment:
This solution creates unnecessary looping when not all 26 letters are used in a word which is pretty often. Any thoughts on how to optimize?